src/cares_wrap.cc

```diff
@@ -1411,12 +1411,13 @@ static void Query(const FunctionCallbackInfo<Value>& args) {
   CHECK(args[1]->IsString());
 
   Local<Object> req_wrap_obj = args[0].As<Object>();
-  Local<String> string = args[1].As<String>();
   auto wrap = std::make_unique<Wrap>(channel, req_wrap_obj);
 
-  node::Utf8Value name(env->isolate(), string);
+  auto plain_name =
```

Comment:
I am not sure this code is safe because it appears that temporary objects are created, and they go out of scope. The result of node::Utf8Value is converted to a string_view, but a string_view pointing at what instance? It is unclear to me.

I recommend breaking this down...

```
Local<String> string = args[1].As<String>();
node::Utf8Value utf8name(env->isolate(), string);
std::string name = ada::idna::to_ascii(utf8name.ToStringView());
```

Reply:
@lemire Would you like to takeover this pull request? Unfortunately, I can't complete it soon.

Reply:
https://github.com/nodejs/node/pull/48354